dist/tools/openocd/openocd.sh

```diff
@@ -0,0 +1,177 @@
+#
+# debug:        starts OpenOCD as GDB server in the background and
+#               connects to the server with the GDB client specified by
+#               the board (DBG environment variable)
+#
+#               options:
+#               GDB_PORT:       port opened for GDB connections
+#               TCL_PORT:       port opened for TCL connections
+#               TELNET_PORT:    port opened for telnet connections
+#
+# debug-server: starts OpenOCD as GDB server, but does not connect to
+#               to it with any frontend. This might be useful when using
+#               IDEs.
+#
+#               options:
+#               GDB_PORT:       port opened for GDB connections
+#               TCL_PORT:       port opened for TCL connections
+#               TELNET_PORT:    port opened for telnet connections
+#
+# reset:        triggers a hardware reset of the target board
+#
+#
+# @author       Hauke Peteresen <hauke.petersen@fu-berlin.de>
+
+# default GDB port
+_GDB_PORT=3333
+# default telnet port
+_TELNET_PORT=4444
+# default TCL port
+_TCL_PORT=6333
+# path to OpenOCD configuration file
+CONFIG=${RIOTBOARD}/${BOARD}/dist/openocd.cfg
+
+#
+# a couple of tests for certain configuration options
+#
+test_config() {
+    if [ ! -f ${CONFIG} ]; then
+        echo "Error: Unable to locate OpenOCD configuration file"
+        echo "       (${CONFIG})"
+        exit 1
+    fi
+}
+
+test_hexfile() {
+    if [ ! -f ${HEXFILE} ]; then
+        echo "Error: Unable to locate HEXFILE"
+        echo "       (${HEXFILE})"
+        exit 1
+    fi
+}
+
+test_ports() {
+    if [ -z ${GDB_PORT} ]; then
+        GDB_PORT=${_GDB_PORT}
+    fi
+    if [ -z ${TELNET_PORT} ]; then
+        TELNET_PORT=${_TELNET_PORT}
+    fi
+    if [ -z ${TCL_PORT} ]; then
+        TCL_PORT=${_TCL_PORT}
+    fi
+}
+
+test_tui() {
+    if [ ! -z ${TUI} ]; then
+        echo "hello hello"
+        TUI=-tui
+    fi
+}
+
+#
+# now comes the actual actions
+#
+do_flash() {
+    test_config
+    test_hexfile
+    # flash device
+    openocd -f ${CONFIG} \
+            -c "tcl_port 0" \
+            -c "telnet_port 0" \
+            -c "gdb_port 0" \
+            -c "init" \
+            -c "targets" \
+            -c "reset halt" \
+            -c "program ${HEXFILE} verify" \
+            -c "reset run"
+}
+
+do_debug() {
+    test_config
+    test_ports
+    test_tui
+    # write temporary gdb command file
+    echo "target extended-remote :${GDB_PORT}" > ${BINDIR}gdb.cfg
+    # start OpenOCD as GDB server
+    openocd -f ${CONFIG} \
+            -c "tcl_port ${TCL_PORT}" \
+            -c "telnet_port ${TELNET_PORT}" \
+            -c "gdb_port ${GDB_PORT}" \
+            -c "init" \
+            -c "targets" \
+            -c "reset halt" \
+            -l /dev/null &
+    # save PID for terminating the server afterwards
+    OCD_PID=$?
+    # connect to the GDB server
+    ${DBG} ${TUI} -command=${BINDIR}/gdb.cfg ${ELFFILE}
```

Comment:
I wonder if that is really in scope of RIOT's `make debug` target. If you want to do some crazy things with gdb, just use `make debug-server` and then start your gdb frontend in the way you want...